.github/workflows/ci.yml

```diff
@@ -0,0 +1,89 @@
+name: CI workflow
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14.x
+
+      - name: Cache npm packages
+        uses: actions/cache@v2
+        id: npm-cache
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+
+      - name: Install dependencies
+        run: npm ci
+        env:
+          CI: true
+
+      - name: Format files
+        run: npm run format
+
+      - name: Commit formatting changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Apply formatting changes
+          branch: ${{ github.head_ref }}
```

Comment:
Maybe this should be in a seperate format job instead of in the lint job?

Reply:
As ESLint used Prettier as its rules, I think it can be done step by step. First formatting all selected file types, then linting JS and TS files. Let's see how it works in the next PRs, I'm only in favour of something to improve if it's going to be inconvenient.

Reply:
Can we try then? or do you want to merge as is?

Reply:
The problem that can arise if we separate formatting and linting is that it will be parallel and not under the control of what happens in the first place. ESLint also runs the Prettier as rules so it can get messy That's why I think you can merge as it is and we can test it in the next PRs as it's not testable right now. Thanks!